src/VisualStudio/Core/Def/Implementation/Library/ObjectBrowser/AbstractObjectBrowserLibraryManager.cs

```diff
@@ -101,6 +96,21 @@ private void OnWorkspaceChanged(object sender, WorkspaceChangeEventArgs e)
             }
         }
 
+        private async Task DocumentChangedAsync(Document oldDocument, Document newDocument)
+        {
+            var oldTextVersion = await oldDocument.GetTextVersionAsync(CancellationToken.None).ConfigureAwait(false);
+            var newTextVersion = await newDocument.GetTextVersionAsync(CancellationToken.None).ConfigureAwait(false);
+
+            if (oldTextVersion != newTextVersion)
+            {
+                lock (classMemberGate)
+                {
+                    UpdateClassVersion();
+                    UpdateMembersVersion();
+                }
+            }
```

Comment:
Given that `UpdateClassVersion` and `UpdateMemberVersion` are only called from here, I would unify them into `UpdateVersions` and put the lock inside that.  Otherwise people look at `Update?Version` and don't realize that the state in it needs to be synchronized.